src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -1600,15 +1638,28 @@ static int freeze_epilog(JavaThread* thread, ContinuationWrapper& cont, freeze_r
   }
 
   JVMTI_ONLY(jvmti_yield_cleanup(thread, cont)); // can safepoint
-  return freeze_epilog(thread, cont);
+  return freeze_epilog(cont);
 }
 
-template<typename ConfigT>
+static int preempt_epilog(ContinuationWrapper& cont, freeze_result res, frame& old_last_frame) {
+  if (UNLIKELY(res != freeze_ok)) {
+    verify_continuation(cont.continuation());
+    log_develop_trace(continuations)("=== end of freeze (fail %d)", res);
+    return res;
+  }
+
+  patch_return_pc_with_preempt_stub(old_last_frame);
+  cont.tail()->set_preempted(true);
+
+  return freeze_epilog(cont);
+}
+
+template<typename ConfigT, bool preempt>
```

Comment:
I checked and release build is same size and fast/slow debug difference is only about 16kb. Since it doesn’t hurt I would rather not touch the fast paths, but I see `ConfigT` has been unused for some time now so I can do a follow up cleanup.